python/graphstorm/inference/ep_infer.py

```diff
@@ -119,10 +124,20 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
 
             # The order of the ntypes must be sorted
             embs = {ntype: embs[ntype] for ntype in sorted(target_ntypes)}
-            save_gsgnn_embeddings(save_embed_path, embs, self.rank,
-                get_world_size(),
-                device=device,
-                node_id_mapping_file=node_id_mapping_file)
+
+            if not saved_hdf5_embed:
+                save_gsgnn_embeddings(save_embed_path, embs, self.rank,
+                    get_world_size(),
+                    device=device,
+                    node_id_mapping_file=node_id_mapping_file)
+            else:
+                # remap gnn embeddings without writing to disk
+                mapped_embeds = remap_embeddings(embeddings, self.rank, get_world_size(),
+                    node_id_mapping_file, device=device)
+                if self.rank == 0:
+                    sys_tracker.check(f"Writing GNN embeddings to {os.path.join(save_embed_path, 'embed_dict.hdf5')}")
+                    os.makedirs(save_embed_path, exist_ok=True)
+                    streamly_write_hdf5_from_dist(mapped_embeds, os.path.join(save_embed_path, "embed_dict.hdf5"))
```

Comment:
Is this implementation scalable?

Reply:
yes, it writes chunk by chunk.

Reply:
But you need all the embeddings in mapped_embeds

Reply:
mapped_embeds is dist tensor.